base/pointer.jl

```diff
@@ -111,8 +118,12 @@ Store a value of type `T` to the address of the `i`th element (1-indexed) starti
 This is equivalent to the C expression `p[i-1] = x`.
 
 The `unsafe` prefix on this function indicates that no validation is performed on the
-pointer `p` to ensure that it is valid. Incorrect usage may corrupt or segfault your
-program, in the same manner as C.
+pointer `p` to ensure that it is valid. Like C, the programmer is responsible for ensuring
+that referenced memory is not freed or garbage collected while invoking this function.
+Incorrect usage may segfault your program. Unlike C, storing memory region allocated as
+different type may be valid provided that that the types are compatible (e.g., `p` has the
+correct alignment, the stored bit pattern is valid for later safe and unsafe loads, `T` does
+not contain any GC-managed objects).
```

Comment:
```suggestion
different type may be valid provided that that the types are compatible.
```

Reply:
Don't you need that `T` does not contain a GC-managed object? I'd imagine storing `(1.0, [1])` to the location allocated as `Tuple{Int64,Vector{Int}}` is "compatible" in terms of the underlying bytes. But I thought I can't do it since I'd miss the write barrier. It's probably cleaner to mention it separately, though.

Reply:
It is valid, as long as the the memory you are storing to is not managed by the Julia GC ("compatible")

Reply:
Ah, I see. I can store pointers to Julia objects to `malloc`'ed memory as long as I root them somewhere while I try to use them as Julia objects.